CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+
+- The **View** class manages the `log-view` and [filters web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component). The log-view webcomponent is responsible for rendering the logs collected (currently in a table form) and managing the log-view context menu. The filters web componenrs are responsible for the UI elements related to the log filters.
+
+- The **Controller** class make sure that the data flow in Model and View are synchronized. Whenever the Model updates with new logs or filter options, the Controller make sure that the View also updates accordingly. It also communicates with the background via `runtime.sendMessage` API.
+
+When the Activity Log page is opened via popup, it fetches the existing logs (if logs were collected before) from the background. While the Activity Log page is opened, it receives real-time logs from background and render them. It can send instructions to the background via `runtime.sendMessage` API sto save logs to a JSON file, load logs from JSON file and clear logs.
+
+#### Activity Log Page - Devtools (Extension Page)
+
+This page can be accessed via "Extension Activity" panel in devtools. The "Extension Activity" devtools panel contains an instance of the Activity Log page where every real time activity log collected is automatically filtered by tab id. The devtools panel retrieves the tab id filter from the devtools panel url search params, set by the devtools page by retrieving it using `browser.devtools.inspectedWindow.tabId` when the devtools panel is being registered.
+
+### Core Features
+
+#### Collecting Logs
+
+The extension receives each activity log in the form of an `object` from activityLog API. The activityLog API schema can been found [here](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json). The background page (backend part) subscribes the `browser.activityLog.onExtensionActivity` API event listener, which gets triggered with an activity log object passed as a parameter everytime an activity is found from any monitored extension. While monitoring other extensions, any newly installed extension is monitored automatically.
+
+- ##### Live Logging
+
+  The real-time logs are collected in the background and send to Activity Log page while it is opened. The extension uses `runtime.sendMessage` API to communicate between background and Activity Log page. The [`sendLogs`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L25-L33) method is responsible for sending logs to Activity Log page. The Activity Log page listens for logs via [`runtime.onMessage`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-activitylog.js#L253-L265) event and render those in the [`log-view`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/).
```

Comment:
Nit: s/background and send to/background and sent to/

Reply:
thanks, fixed :-)